include/circt/Dialect/HW/InstanceGraphBase.h

```diff
@@ -183,21 +183,25 @@ class InstanceGraphBase {
   virtual ~InstanceGraphBase();
 
   /// Look up an InstanceGraphNode for a module.
-  InstanceGraphNode *lookup(HWModuleLike op);
+  template<typename T>
+  InstanceGraphNode *lookup(T op) {
```

Comment:
This seems a little strange and unnecessary since presumably everything you lookup is already an Instantiable (and if it's not, shouldn't be trying to look it up anyway)?

Appears that by virtue of `SymbolOpInterface` it gets the `getName()` (used later in this file) and `getNameAttr()` methods...

Reply:
The main thing that was bugging me, and I'll probably walk this back is that given the inheritance
`FModuleLike -> Instantiable` in the interfaces, you cannot cast from FModuleLike to Instantiable, so you have to add casts at all the sites.

Reply:
Or maybe I had just messed up.  Checking that now.